src/main/java/me/jellysquid/mods/lithium/mixin/gen/features/StructureFeatureMixin.java

```diff
@@ -0,0 +1,70 @@
+package me.jellysquid.mods.lithium.mixin.gen.features;
+
+import net.minecraft.structure.StructureStart;
+import net.minecraft.util.math.ChunkPos;
+import net.minecraft.util.math.ChunkSectionPos;
+import net.minecraft.world.StructureHolder;
+import net.minecraft.world.WorldView;
+import net.minecraft.world.chunk.Chunk;
+import net.minecraft.world.chunk.ChunkStatus;
+import net.minecraft.world.gen.StructureAccessor;
+import net.minecraft.world.gen.feature.StructureFeature;
+import org.objectweb.asm.Opcodes;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.asm.mixin.injection.Slice;
+
+@Mixin(StructureFeature.class)
+public class StructureFeatureMixin {
+    /**
+     * Why generate an empty chunk to check for a structure if
+     * the chunk's biome cannot generate the structure anyway?
+     * Checking the biome first = SPEED!
+     *
+     * @author TelepathicGrunt
+     */
+
+    /**
+     * @reason Return null chunk if biome doesn't match structure
+     * @author MrGrim
+     */
+    @Redirect(method = "locateStructure", at = @At(value = "INVOKE", target = "Lnet/minecraft/world/WorldView;getChunk(IILnet/minecraft/world/chunk/ChunkStatus;)Lnet/minecraft/world/chunk/Chunk;", ordinal = 0),
+            slice = @Slice(from = @At(value = "FIELD", opcode = Opcodes.GETSTATIC, target = "Lnet/minecraft/world/chunk/ChunkStatus;STRUCTURE_STARTS:Lnet/minecraft/world/chunk/ChunkStatus;", ordinal = 0),
+                    to = @At(value = "INVOKE", target = "Lnet/minecraft/world/chunk/Chunk;getPos()Lnet/minecraft/util/math/ChunkPos;", ordinal = 0)))
+    private Chunk biomeConditionalGetChunk(WorldView worldView, int x, int z, ChunkStatus status)
+    {
+        if (worldView.getBiomeForNoiseGen((x << 2) + 2, 0, (z << 2) + 2).hasStructureFeature((StructureFeature) (Object) this))
```

Comment:
After rebasing onto the 1.16.2 dev branch, the hasStructureFeature method does not seem to exist. How can this be fixed?

Reply:
It’s now 
worldView.getBiomeForNoiseGen((chunkPos.x << 2) + 2, 60, (chunkPos.z << 2) + 2).getGenerationSettings().hasStructureFeature((StructureFeature)(Object)this)

Reply:
Okay thank you, but where does that 60 come from? ChunkGenerator.setStructureStarts doesn't have it

Reply:
The y value doesn’t matter. I just chose 60 as default but if you follow getBiomeForNoiseGen, it’ll just use the x and z values only if I recall. Keep it 0 for now

Reply:
Actually, in the nether the generator uses x, z, *and* y. I think 0 would be a better choice here, to keep a good baseline.

Reply:
In the code I found calls with y=0, y=2, y = getSeaLevel(). Structure code seems to use 0